### PR TITLE
gc-featured-link: adding bg color option

### PR DESCRIPTION
**Pull request by Garneauma**
Adding option to customize the background colour of featured links.

Added CSS to hide the customized link by default to prevent a colour flick on load.

### CODE REVIEW

**Review by GormFrank — CHANGES_REQUESTED** [review]
Partial review

File: components/gc-featured-link/gc-featured-link.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		// returns DOM object = proceed with init
+		// returns undefined = do not proceed with init (e.g., already initialized)
+		var elm = wb.init( event, componentName, selector ),
+			$elm;
+
+		if ( elm ) {
+			$elm = $( elm );
+
+			elm.style.backgroundColor = elm.dataset.bgColor;
+			elm.style.display = "block";
```
[the remark this answers is not in the document]
Comment (Garneauma) [reply components/gc-featured-link/gc-featured-link.js]: Great suggestion. Done.

**Review by duboisp — CHANGES_REQUESTED** [review]
* [ ] This is broken in basic HTML mode, we should still display the featured banner. In basic mode I can see a white text with a white background.
* [ ] How do we ensure the text inside the banner do have and maintain a sufficient contrast ratio? Should we auto switch the text color between black and white depending of the luminosity of the background color that was set?

File: components/gc-featured-link/_base.scss [hunk 2]
```diff
@@ -17,5 +17,9 @@
 		p, a {
 			color: #fff;
 		}
+
+		html:not(.wb-disable) &[data-bg-color] {
+			background-color: transparent;
```
Comment (duboisp) [inline components/gc-featured-link/_base.scss:22]: We would need to set a default text color which we do know its going to be rewritten when the plugin are initialized.

File: components/gc-featured-link/gc-featured-link-en.html [hunk 3]
```diff
@@ -52,17 +52,104 @@ <h2 class="wb-inv">Featured</h2>
...
 		&lt;/div&gt;
-	&lt;/section&gt;
-	</code></pre>
+	&lt;/div&gt;
+&lt;/section&gt;</code></pre>
+
+	<h3>Customized background color</h3>
+	<p>To customize the background color, add the attribute <code>data-bg-color</code>. The value should be the hexadecimal code of the desired color.</p>
+	<p>The plugin will automatically apply the correct color to the text: <code>#FFFFFF</code> for dark colors, <code>#333333</code> for light colors, <code>#000000</code> for colors in-between.</p>
+	<p class="mrgn-tp-lg">Dark background example:</p>
```
Comment (duboisp) [inline components/gc-featured-link/gc-featured-link-en.html:69]: We would need to add a notice about the irregular heading structure along with a skip link to the code sample.

File: components/gc-featured-link/gc-featured-link-en.html [hunk 4]
```diff
@@ -52,17 +52,104 @@ <h2 class="wb-inv">Featured</h2>
...
+				<h2 class="wb-inv">Featured</h2>
+				<p><a class="stretched-link" href="#">[Promotion title]</a></p>
+				<p>Long description [Optional]</p>
+			</div>
+		</div>
+	</div>
+</section>
+
+<div class="container">
+	<h3>Code</h3>
```
Comment (duboisp) [inline components/gc-featured-link/gc-featured-link-en.html:117]: Should this be an heading 4 considering this is a subsection of "Customized background color"

**Review by duboisp — CHANGES_REQUESTED** [review]
I love the way it is setup and do self-adjust to ensure it meet the contrast ratio.

File: components/gc-featured-link/_base.scss [hunk 5]
```diff
@@ -1,21 +1,24 @@
 /* Provisional featured link component */
 
-.gc-featured-link {
+.gc-featured-link.provisional {
```
Comment (duboisp) [inline components/gc-featured-link/_base.scss:3]: ceci nous avais causé quelque problème d'override. on va devoir utiliser une nouvelle stratégie pour identifier les provisional feature et les retirer/neutralisé le cas échéant.

Enlève la classe `.provisional`

File: components/gc-featured-link/_base.scss [hunk 6]
```diff
@@ -1,21 +1,24 @@
...
+		color: #fff;
+		font-weight: $bold-weight;
+	}
 
-		p, a {
-			color: #fff;
-		}
+	html:not(.wb-disable) &[data-bg-color] {
+		background-color: transparent;
+		color: #fff;
```
Comment (duboisp) [inline components/gc-featured-link/_base.scss:22]: Considérant que notre background est white-smoke je crois que la couleur de text adéquat serait noir et non blanc.

```suggestion
		color: #000;
```

Et un override additionnelle sera requis pour les ancres

```suggestion
		color: #000;
		
		a {
			color: #000;
		}
```

**Review by duboisp — CHANGES_REQUESTED** [review]
Testé localement et ça fonctionne tel que prévu, il va rester à adjuster la couleur du texte par défaut juste avant que l'initialization soit complété.

**Review by duboisp — COMMENTED** [review]
Pre-approved upon the default loading style in standard mode do have sufficient contrast ratio

**Review by duboisp — APPROVED** [review]
Tested locally and reviewed. Every work as expected.

**Pull request merged**